proxy/envoy/egress.go

```diff
@@ -26,113 +28,60 @@ import (
 	"istio.io/manager/proxy"
 )
 
+const (
+	egressNode = "egress"
+)
+
 type egressWatcher struct {
-	agent   proxy.Agent
-	ctl     model.Controller
-	context *EgressConfig
+	agent proxy.Agent
+	mesh  *config.ProxyMeshConfig
 }
 
 // NewEgressWatcher creates a new egress watcher instance with an agent
-func NewEgressWatcher(ctl model.Controller, context *EgressConfig) (Watcher, error) {
-	agent := proxy.NewAgent(runEnvoy(context.Mesh, "egress"), cleanupEnvoy(context.Mesh), 10, 100*time.Millisecond)
-
-	out := &egressWatcher{
-		agent:   agent,
-		ctl:     ctl,
-		context: context,
+func NewEgressWatcher(mesh *config.ProxyMeshConfig) (Watcher, error) {
+	if mesh.EgressProxyAddress == "" {
+		return nil, errors.New("egress proxy requires address configuration")
 	}
-
-	// egress depends on the external services declaration being up to date
-	if err := ctl.AppendServiceHandler(func(*model.Service, model.Event) {
-		out.reload()
-	}); err != nil {
-		return nil, err
-	}
-
-	return out, nil
-}
-
-func (w *egressWatcher) reload() {
-	w.agent.ScheduleConfigUpdate(generateEgress(w.context))
+	agent := proxy.NewAgent(runEnvoy(mesh, egressNode), cleanupEnvoy(mesh), 10, 100*time.Millisecond)
+	return &egressWatcher{
+		agent: agent,
+		mesh:  mesh,
+	}, nil
 }
 
 func (w *egressWatcher) Run(stop <-chan struct{}) {
 	go w.agent.Run(stop)
+	w.agent.ScheduleConfigUpdate(generateEgress(w.mesh))
+	<-stop
+}
 
-	// Initialize envoy according to the current model state,
-	// instead of waiting for the first event to arrive.
-	// Note that this is currently done synchronously (blocking),
-	// to avoid racing with controller events lurking around the corner.
-	// This can be improved once we switch to a mechanism where reloads
-	// are linearized (e.g., by a single goroutine reloader).
-	w.reload()
-	w.ctl.Run(stop)
+func getEgressProxyPort(mesh *config.ProxyMeshConfig) int {
+	addr := mesh.EgressProxyAddress
+	port, _ := strconv.Atoi(addr[strings.Index(addr, ":")+1:])
```

Comment:
Just to confirm: we verify the addr format in the validation process so it's safe to ignore the error here?